src/Microsoft.AspNetCore.Mvc.RazorPages/ApplicationModels/PageConventionCollection.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    public class PageConventionCollection : Collection<IPageConvention>
+    {
+        public IPageApplicationModelConvention AddConvention(string pageName, Action<PageApplicationModel> action)
```

Comment:
This kind of overloading `Action<PageApplicationModel> action` vs `Action<PageRouteModel> action` is not good. These need to have different names